### ISSUE TITLE
3.15.x releases seem to break client-side navigation

### ISSUE THREAD

**Issue by mbice** [microsoft-search/pnp-modern-search#493]
**Version used**
3.15.2

**Describe the bug**
This issue is not present in 3.14.2. Upon upgrading to 3.15.x I noticed that when navigating to a page with the PnP search web parts on it, subsequent navigation to other sites breaks.

**To Reproduce**
1. Install 3.15.2 package to tenant app catalog and globally deploy
2. Add search box and search results web parts to a page
3. Link up the search box to the results web part, set default query to "*". Leave all other config default.
4. Load up the page. Search results show up fine.
5. Navigate to another site, it loads fine
6. Navigate once more to another site and the client-side navigation is broken. Clicking sites in the hub nav changes the URL, but the page doesn't update. You need to fully reload the page to get navigation working again.

**Screenshots**
![image](https://user-images.githubusercontent.com/985283/94039310-688c0400-fd7c-11ea-8688-dfc32389affc.png)

**Desktop:**
 - Windows 10, Chrome 85.0.4183.102

**Comment by wobba** [comment]
5 and 6, how do I navigate to a new site? Where do I click? What's the navigation setup?

**Comment by mbice** [comment]
@wobba I’m just navigating using out-of-the-box hub navigation. I have a handful of sites connected to the root site, which is registered as a hub site.

Thanks!

**Comment by wobba** [comment]
Hub site, got it. Will try to repro.

**Comment by mbice** [comment]
@wobba just wanted to check in. Were you able to reproduce this on your end?

Thanks!

**Comment by wobba** [comment]
Haven't had any spare time for this and was oof last week.

**Comment by MakingThingsWorkToronto** [comment]
@mbice - is your tenant on Standard Release?

**Comment by mbice** [comment]
@MakingThingsWorkToronto I just re-tested with the latest release (3.15.3) and it reproduces fully on a standard release tenant. On targeted release I'm seeing the same error, albeit less frequently and in a different way. On targeted release I am able to reproduce it with the following steps:

1. Edit a page and add search box and search results web parts (default configuration)
2. Publish page
3. Try to navigate away using hub navigation -- same error as the one I originally logged

**Comment by wobba** [comment]
Pushed a fix to the dev branch. @mbice are you able to build this one and verify?

**Comment by mbice** [comment]
Hi @wobba I rebuilt and tried again on both standard and targeted release tenants. I'm still seeing the issues I've outlined. The bug is less prevalent on targeted release, where it only happens after publishing a page with search web parts on them. On standard all I need to do to reproduce is navigate to the page with the web parts, then navigate away and it breaks navigation entirely.

![image](https://user-images.githubusercontent.com/985283/96901451-2243c680-1448-11eb-956a-fef6d14c02a6.png)

Seems to be failing on this call.

![image](https://user-images.githubusercontent.com/985283/96901862-a433ef80-1448-11eb-925e-cbf131bf47a2.png)

Thanks!

**Comment by wobba** [comment]
@mbice I do see that one as well when I edit/save etc... Logged https://github.com/SharePoint/sp-dev-docs/issues/6403 which I'm curios if can fix this. If we are to add checks all over, that's a lot of checks.

**Comment by saregbesola** [comment]
@wobba, we are also experiencing the same issue. I just upgraded to version 3.15.3 to test the fix provided by pull request #394 that fixes issue #302. I confirmed that is being fixed but  it seems the fix brought about the behavior @mbice described here which seems to be the opposite of what we were experiencing with version 3.14.x. I just wanted to provide more info...perhaps it will be helpful.
